metatron/src/main/java/uk/gov/ida/eidas/metatron/domain/MetadataResolverService.java

```diff
@@ -58,5 +93,74 @@ private MetadataResolver createMetadataResolver(EidasCountryConfig country) {
                 MAX_REFRESH_DELAY_MS);
         return resolver;
     }
-}
 
+    private EidasCountryConfig getCountryConfig(URI entityId) {
+        EidasCountryConfig eidasCountryConfig = metadataResolverMap.keySet().stream()
+                .filter(c -> entityId.equals(c.getEntityId()))
+                .findFirst()
+                .orElseThrow(() -> new MetatronClientException(String.format("No metadata entry for entityID %s", entityId)));
+        if (!eidasCountryConfig.isEnabled()) {
+            throw new MetatronClientException(String.format("EntityID %s is not enabled", entityId));
+        }
+        return eidasCountryConfig;
+    }
+
+    private MetadataResolver getMetadataResolver(EidasCountryConfig eidasCountryConfig) {
+        return metadataResolverMap.get(eidasCountryConfig);
+    }
+
+    private String getCertificateAsX509(EidasCountryConfig countryConfig, UsageType usageType) {
+        X509Credential credential = (X509Credential) getCredential(usageType, countryConfig, SPSSODescriptor.DEFAULT_ELEMENT_NAME);
```

Comment:
As the `DEFAULT_ELEMENT_NAME` is an imported constant you could use it directly in `getCredential` rather than have it as a param.

Reply:
Yes, done